packages/@aws-cdk/aws-appmesh/lib/tls-validation.ts

```diff
@@ -125,11 +164,110 @@ class TlsValidationFileTrust extends TlsValidationTrust {
           certificateChain: this.certificateChain,
         },
       },
+      virtualNodeListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
       virtualGatewayClientTlsValidationTrust: {
         file: {
           certificateChain: this.certificateChain,
         },
       },
+      virtualGatewayListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
     };
   }
 }
+
+class TlsValidationSdsTrust extends TlsValidationTrust {
+  /**
+   * The name of the secret for Envoy to fetch from a specific endpoint through the Secrets Discovery Protocol.
+   */
+  readonly secretName: string;
+
+  constructor (props: TlsValidationSdsTrustOptions) {
+    super();
+    this.secretName = props.secretName;
+  }
+
+  public bind(_scope: Construct): TlsValidationTrustConfig {
+    return {
+      virtualNodeClientTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualNodeListenerTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualGatewayClientTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualGatewayListenerTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+    };
+  }
+}
+
+/**
+ * All Properties for Subject Alternative Names Matcher for both Client Policy and Listener.
+ */
+export interface SubjectiveAlternativeNamesMatcherConfig {
+
+  /**
+   * VirtualNode CFN configuration for subject alternative names secured by the certificate.
+   */
+  readonly virtualNodeSans: CfnVirtualGateway.SubjectAlternativeNameMatchersProperty;
+
+  /**
+   * VirtualGateway CFN configuration for subject alternative names secured by the certificate.
+   */
+  readonly virtualGatewaySans: CfnVirtualGateway.SubjectAlternativeNameMatchersProperty;
+}
+
+/**
+ * Used to generate Subject Alternative Names Matchers
+ */
+export abstract class SubjectiveAlternativeNamesMatch {
+  /**
+   * The values of the SAN must match the specified values exactly.
+   *
+   * @param subjectAlternativeNames The exact values to test against.
+   */
+  public static valuesAre(subjectAlternativeNames: string[]): SubjectiveAlternativeNamesMatch {
```

Comment:
IMO we should rename this to `exact` because it's more explicit.

```
subjectAlternativeNames: appmesh.SubjectiveAlternativeNamesMatch.exactMatch(['san1', 'san2']),
```

Another consideration: Currently there is only one type of match (exact) but in the future we could have more. It's possible we'll allow users to define multiple match modes which would mean we'd have to break this part of the CDK API when we update it.

Not using a factory here could be more safe. Like this:

```
subjectAlternativeNames:
    exactMatch: []
    prefixMatch: [] # future?
```

 Thoughts? + @dfezzie

Reply:
+1 to your suggestions. I prefer `exact` over `valuesAre`

As long as it's semantically valid to have multiple different matches, that seems right.

Reply:
Good point AJ. I agree with you avoiding to use the factory method is safe option at the moment.